test/libsolidity/smtCheckerTests/operators/constant_propagation_1.sol

```diff
@@ -0,0 +1,13 @@
+pragma experimental SMTChecker;
+
+contract C {
+	uint constant DEPOSIT_CONTRACT_TREE_DEPTH = 32;
+	uint constant MAX_DEPOSIT_COUNT = 2**DEPOSIT_CONTRACT_TREE_DEPTH - 1;
+	function f() public pure {
+		assert(DEPOSIT_CONTRACT_TREE_DEPTH == 32);
+		assert(MAX_DEPOSIT_COUNT == 4294967295);
```

Comment:
Note that these only work for CHC. BMC does assign `uint constant MAX_DEPOSIT_COUNT = 4294967295` properly, but it forgets that when analyzing functions separately, since it doesn't have state knowledge. One solution ofc could be to explicitly save values of constant variables as they were constructed, specifically for BMC.

Reply:
I still wonder why don't we just have an annotation on constant variables telling the evaluated values?

Reply:
That I don't know